lib/middleware.js

```diff
@@ -86,11 +87,25 @@ module.exports = function(watcher) {
       response.setHeader('Content-Length', stat.size)
       response.setHeader('Content-Type', type)
 
-      // read file sync so we don't hold open the file creating a race with
-      // the builder (Windows does not allow us to delete while the file is open).
-      buffer = fs.readFileSync(filename)
-      response.writeHead(200)
-      response.end(buffer)
+      function writeResponse() {
+        // read file sync so we don't hold open the file creating a race with
+        // the builder (Windows does not allow us to delete while the file is open).
+        buffer = fs.readFileSync(filename)
+        response.writeHead(200)
+        response.end(buffer)
+      }
+
+      if(request.headers.origin) {
+        // Check if origin resolves to a loopback IP, only add CORS headers if that's the case.
+        dns.lookup(url.parse(request.headers.origin).hostname, function(err, address) {
```

Comment:
The DNS lookup is convenient for people who run their dev servers on a local domain name (think /etc/hosts). Something like [pow](http://pow.cx/) would work with this DNS lookup, whereas a simple string comparison wouldn't work. Maybe that's acceptable though, it would definitely make for prettier code.